nominal/core/channel.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import dataclasses
+import enum
+
+from typing_extensions import Self
+
+from .._api.combined import datasource_api, timeseries_logicalseries_api
+from ._utils import HasRid
+
+
+class ChannelDataType(enum.Enum):
+    DOUBLE = "DOUBLE"
+    STRING = "STRING"
+    UNKNOWN = "UNKNOWN"
+
+    @classmethod
+    def _from_conjure(
+        cls, data_type: datasource_api.SeriesDataType | timeseries_logicalseries_api.SeriesDataType
+    ) -> Self:
+        if data_type.value in cls:
+            return cls(data_type.value)
+        else:
+            return cls("UNKNOWN")
+
+
+@dataclasses.dataclass
+class Channel(HasRid):
+    """Metadata for working with channels."""
+
+    rid: str
+    name: str
+    data_source: str
+    data_type: ChannelDataType | None
```

Comment:
when does a channel not have a data type btw?

Reply:
I would like to investigate this more (in the future)...seems kinda incorrect? maybe its for before ingestion has completed?